metrics/cloudwatch/cloudwatch.go

```diff
@@ -102,8 +104,8 @@ func New(namespace string, svc cloudwatchiface.CloudWatchAPI, options ...option)
 		percentiles:           []float64{0.50, 0.90, 0.95, 0.99},
 	}
 
-	for _, optFunc := range options {
-		optFunc(cw)
+	for _, opt := range options {
+		cw.apply(opt)
 	}
```

Comment:
Can we just remove the apply method altogether?

Reply:
It's actually more safer than a simple function. `apply` cannot be called outside of the package, but with the current `Option` type it's possible to modify the state of `Cloudwatch` after the initialization.

I'd instead change the signature of `Option` to this:

```go
type Option interface{
    apply(*CloudWatch)
}
```

Reply:
> I'd instead change the signature of Option to this:

Strong -1. Option interfaces are an antipattern, and especially those that take this form, i.e. an exported interface with an unexported method.

> with the current Option type it's possible to modify the state of Cloudwatch after the initialization.

I understand your point, but I don't think this is a meaningful risk.

Reply:
I think we can remove the `apply` method, because `cloudwatch.New(namespace, svc, nil)` isn't norm.

Reply:
fixed